### PR TITLE
Feat/maker ldk

### PR DESCRIPTION
**Pull request by da-kami**

### CODE REVIEW

**Review by klochowicz — APPROVED** [review]
contents looks good.
nit: the commits might as well be squashed together (except using regtest and nigiri, the first one), because the split doesn't work too well (there's a few unrelated changes, and then a commit called "fix formatting").

File: rust/src/lightning.rs [hunk 1]
```diff
@@ -525,6 +528,110 @@ pub async fn run_ldk(system: &LightningSystem) -> Result<BackgroundProcessor> {
     Ok(background_processor)
 }
 
+pub async fn run_ldk_server(
```
Comment (klochowicz) [inline rust/src/lightning.rs:531]: I suspect we could just use `run_ldk` on both sides (and just listen for incoming connections on both). Two different functions, doing almost the same, make things more confusing than actually solve.

Comment (klochowicz) [reply rust/src/lightning.rs:532]: in the meantime, to make things more maintainable, I just refactored the code a bit (so it calls common path afterwards) - this should make it much less confusing.

File: rust/src/lightning.rs [hunk 2]
```diff
@@ -558,77 +558,7 @@ pub async fn run_ldk_server(
...
-                        &PeerInfo { pubkey, peer_addr },
-                        system.peer_manager.clone(),
-                    )
-                    .await;
-                });
-            }
-        }
-    }
-    tracing::info!("Lightning network started");
+    let background_processor = run_ldk(system).await?;
```
Comment (da-kami) [inline rust/src/lightning.rs:561]: We could extract the listening part into a separate function and have two calls on the server, one for starting the listening and then just `run_ldk`. The `tcp_handle is not used in `run_ldk`.
Not important enough to fix now though :)

**Pull request merged**